pipewire/qubes-pw-module.c

```diff
@@ -1143,19 +1161,36 @@ static void playback_stream_process(void *d)
 
     update_rate(&stream->rm, (uint32_t)ready, pw_stream_is_driving(stream->stream), stream->direction);
 
+    uint32_t to_write;
     if (size > (uint32_t)ready) {
         pw_log_warn("Overrun: asked to write %" PRIu32 " bytes, but can only write %d", size, ready);
         process_control_commands(impl);
-        size = (uint32_t)ready;
+        to_write = (uint32_t)ready;
+    } else {
+        to_write = size;
     }
 
-    pw_log_trace("writing %" PRIu32 " bytes to vchan", size);
-    if (size > 0) {
+    pw_log_trace("writing %" PRIu32 " bytes to vchan", to_write);
+    if (to_write > 0) {
         errno = 0;
-        if (libvchan_send(stream->vchan, data, size) != (int)size)
+        if (libvchan_send(stream->vchan, data, to_write) != (int)to_write)
             pw_log_error("vchan error: %m");
     }
     pw_stream_queue_buffer(stream->stream, buf);
+    if (stream->bytes_before_sending_cmd != 0) {
+        if (size >= stream->bytes_before_sending_cmd) {
+            pw_log_trace("Uncorking stream after trying to send %" PRIu32 " bytes",
+                         size);
+            uint32_t cmd = QUBES_PA_SINK_UNCORK_CMD;
+            if (libvchan_send(stream->impl->stream[PW_DIRECTION_OUTPUT].vchan, &cmd, sizeof(cmd)) != (int)sizeof(cmd))
```

Comment:
wrong vchan?

Reply:
I had used the wrong vchan for a while.  Took a long time to debug.  In this case, `PW_DIRECTION_OUTPUT` is the _PipeWire_ output stream, which corresponds to audio _capture_, so the code is correct.  I will add a comment.